### ISSUE TITLE
Vulnerability detector fails when using <url> tag for NVD

### ISSUE THREAD

**Issue by jctello** [wazuh/wazuh#5154]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.12.3-31209 | Vulnerability Detector | Manager | Packages | CentOS 7.8.2003 |

## Description
When using the `<url>` tag to provide a custom URL for the vulnerability feed of the NVD the system doesn't update more than one year of feeds. 

## Steps to reproduce
After installing the latest version of the Wazuh Manager, follow the [steps in the Wazuh documentation](https://documentation.wazuh.com/3.12/user-manual/capabilities/vulnerability-detection/offline_update.html#national-vulnerability-database) to configure the Offline CVE Feed for NVD. While also uncompressing the feed files and adjusting the configuration accordingly. 

In the `ossec.log` after restarting the database update will start and after some minutes it will state `SQL error: no such table: NVD_CVE_MATCHES` followed by `Could not refresh the NVD DB` and `CVE database could not be updated`. As a result vulnerabilities are not alerted either in full or partially.

The exact commands to reproduce on a new CentOS 7 machine whose IP is `192.168.14.2` are:
```
rpm --import https://packages.wazuh.com/key/GPG-KEY-WAZUH
cat > /etc/yum.repos.d/wazuh.repo <<\EOF
[wazuh_repo]
gpgcheck=1
gpgkey=https://packages.wazuh.com/key/GPG-KEY-WAZUH
enabled=1
name=Wazuh repository
baseurl=https://packages.wazuh.com/3.x/yum/
protect=1
EOF
yum install wazuh-manager-3.12.3 -y
curl --silent --location https://rpm.nodesource.com/setup_10.x | bash -
yum install nodejs -y
yum install wazuh-api-3.12.3 -y
sed -i "s/^enabled=1/enabled=0/" /etc/yum.repos.d/wazuh.repo
yum install epel-release -y
yum install nginx -y
systemctl start nginx
systemctl enable nginx
curl --remote-name https://raw.githubusercontent.com/wazuh/wazuh/v3.12.3/tools/vulnerability-detector/nvd-generator.sh
chmod u+x nvd-generator.sh
mkdir /usr/share/nginx/html/nvd-feed
./nvd-generator.sh 2002 /usr/share/nginx/html/nvd-feed
gunzip /usr/share/nginx/html/nvd-feed/*gz
cat >> /var/ossec/etc/ossec.conf <<\EOF
<ossec_config>
  <vulnerability-detector>
    <enabled>yes</enabled>
    <interval>5m</interval>
    <ignore_time>6h</ignore_time>
    <run_on_start>yes</run_on_start>
    <provider name="nvd">
      <enabled>yes</enabled>
      <url start="2015" end="2020">http://192.168.14.2/nvd-feed/nvd-feed[-].json</url>
      <update_interval>1h</update_interval>
    </provider>
  </vulnerability-detector>
</ossec_config>
EOF
#vi /var/ossec/etc/internal_options.conf # modules.debug=2
systemctl restart wazuh-manager
```

## Additional information
Some times the database will load the first year of information specified in `<url start=`
This can be observed by running the command: 
```
for i in {1999..2020}; do echo $i; sqlite3 /var/ossec/queue/vulnerabilities/cve.db 'select * from NVD_CVE where CVE_ID like "CVE-'$i'-%" limit 1'; done
```

Enabling `wazuh_modules.debug=2` in the `internal_options.conf` will provide the following messages when using `<url start="2019" end="2020">`:
```2020/06/01 22:11:44 wazuh-modulesd[28553] wmodules-vuln-detector.c:708 at wm_vuldet_read_provider(): DEBUG: Added nvd feed. Interval: 3600s | Multi path: 'none' | Multi url: 'http://192.168.14.2/nvd-feed/nvd-feed[-].json' | Update since: 2010.
2020/06/01 22:11:44 wazuh-modulesd[28553] wmodules.c:123 at wm_check(): DEBUG: Deleting repeated module 'vulnerability-detector'.
2020/06/01 22:11:53 wazuh-modulesd[28862] wmodules-vuln-detector.c:708 at wm_vuldet_read_provider(): DEBUG: Added nvd feed. Interval: 3600s | Multi path: 'none' | Multi url: 'http://192.168.14.2/nvd-feed/nvd-feed[-].json' | Update since: 2010.
2020/06/01 22:11:53 wazuh-modulesd[28862] wmodules.c:123 at wm_check(): DEBUG: Deleting repeated module 'vulnerability-detector'.
2020/06/01 22:11:53 wazuh-modulesd[28862] main.c:98 at main(): DEBUG: Created new thread for the 'vulnerability-detector' module.
2020/06/01 22:11:53 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2556 at wm_vuldet_check_feed(): DEBUG: (5461): Starting Wazuh CPE dictionary database update.
2020/06/01 22:11:53 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:3907 at wm_vuldet_fetch_wazuh_cpe(): DEBUG: (5457): Wazuh CPE dictionary OVAL is in its latest version. Update date: 2019-10-02T10:56Z
2020/06/01 22:11:53 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2580 at wm_vuldet_check_feed(): DEBUG: (5461): Starting Wazuh CPE dictionary database update.
2020/06/01 22:11:53 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2556 at wm_vuldet_check_feed(): DEBUG: (5461): Starting Microsoft Security Update database update.
2020/06/01 22:11:54 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2222 at wm_vuldet_index_feed(): DEBUG: (5458): Refreshing Microsoft Security Update databases.
2020/06/01 22:11:54 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1245 at wm_vuldet_insert(): DEBUG: (5473): Inserting vulnerabilities.
2020/06/01 22:11:54 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1272 at wm_vuldet_insert(): DEBUG: (5530): Inserting Microsoft Security Update dictonary.
2020/06/01 22:11:56 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2228 at wm_vuldet_index_feed(): DEBUG: (5459): Refresh of Microsoft Security Update database finished.
2020/06/01 22:11:56 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2233 at wm_vuldet_index_feed(): DEBUG: remove(tmp/vuln-temp): No such file or directory
2020/06/01 22:11:56 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2239 at wm_vuldet_index_feed(): DEBUG: remove(tmp/vuln-temp.bz2): No such file or directory
2020/06/01 22:11:56 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2580 at wm_vuldet_check_feed(): DEBUG: (5461): Starting Microsoft Security Update database update.
2020/06/01 22:11:56 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2554 at wm_vuldet_check_feed(): INFO: (5461): Starting National Vulnerability Database database update.
2020/06/01 22:11:56 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1155 at wm_vuldet_update_feed(): DEBUG: (5512): Synchronizing the year 2019 of the vulnerability database.
2020/06/01 22:11:56 wazuh-modulesd:download[28862] wm_download.c:213 at wm_download_dispatch(): DEBUG: Downloading 'http://192.168.14.2/nvd-feed/nvd-feed2019.json' to '/var/ossec/tmp/vuln-temp-fitted'
2020/06/01 22:15:35 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2222 at wm_vuldet_index_feed(): DEBUG: (5458): Refreshing National Vulnerability Database databases.
2020/06/01 22:15:35 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1245 at wm_vuldet_insert(): DEBUG: (5473): Inserting vulnerabilities.
2020/06/01 22:15:35 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1280 at wm_vuldet_insert(): DEBUG: (5516): Inserting NVD section.
2020/06/01 22:15:41 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:4764 at wm_vuldet_index_nvd(): DEBUG: (5538): It took 6 seconds to index the NVD vulnerabilities.
2020/06/01 22:15:41 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2228 at wm_vuldet_index_feed(): DEBUG: (5459): Refresh of National Vulnerability Database database finished.
2020/06/01 22:15:41 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2233 at wm_vuldet_index_feed(): DEBUG: remove(tmp/vuln-temp): No such file or directory
2020/06/01 22:15:41 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2239 at wm_vuldet_index_feed(): DEBUG: remove(tmp/vuln-temp.bz2): No such file or directory
2020/06/01 22:15:41 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1155 at wm_vuldet_update_feed(): DEBUG: (5512): Synchronizing the year 2020 of the vulnerability database.
2020/06/01 22:15:41 wazuh-modulesd:download[28862] wm_download.c:213 at wm_download_dispatch(): DEBUG: Downloading 'http://192.168.14.2/nvd-feed/nvd-feed2020.json' to '/var/ossec/tmp/vuln-temp-fitted'
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2222 at wm_vuldet_index_feed(): DEBUG: (5458): Refreshing National Vulnerability Database databases.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1245 at wm_vuldet_insert(): DEBUG: (5473): Inserting vulnerabilities.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1280 at wm_vuldet_insert(): DEBUG: (5516): Inserting NVD section.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1095 at wm_vuldet_sql_error(): ERROR: (5405): SQL error: no such table: NVD_CVE_MATCHES
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2225 at wm_vuldet_index_feed(): ERROR: (5409): Could not refresh the NVD DB.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2233 at wm_vuldet_index_feed(): DEBUG: remove(tmp/vuln-temp): No such file or directory
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2239 at wm_vuldet_index_feed(): DEBUG: remove(tmp/vuln-temp.bz2): No such file or directory
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2562 at wm_vuldet_check_feed(): DEBUG: (5437): Failed when updating 'nvd provider' database. Retrying in 300 seconds.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:3061 at wm_vuldet_main(): ERROR: (5426): CVE database could not be updated.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:5175 at wm_vuldet_run_scan(): INFO: (5452): Starting vulnerability scanning.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1060 at wm_vuldet_check_agent_vulnerabilities(): DEBUG: (5486): Starting vulnerability assessment for agent 1.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2846 at wm_vuldet_get_software_info(): DEBUG: (5462): Getting agent 1 software.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2864 at wm_vuldet_get_software_info(): DEBUG: (5574): A partial scan will be run on agent 1.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:3021 at wm_vuldet_get_software_info(): DEBUG: (5475): No changes have been found with respect to the last package inventory or no packages have been indexed for agent 1.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:1060 at wm_vuldet_check_agent_vulnerabilities(): DEBUG: (5486): Starting vulnerability assessment for agent 0.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2846 at wm_vuldet_get_software_info(): DEBUG: (5462): Getting agent 0 software.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:2864 at wm_vuldet_get_software_info(): DEBUG: (5574): A partial scan will be run on agent 0.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:3021 at wm_vuldet_get_software_info(): DEBUG: (5475): No changes have been found with respect to the last package inventory or no packages have been indexed for agent 0.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:5183 at wm_vuldet_run_scan(): INFO: (5453): Vulnerability scanning finished.
2020/06/01 22:16:10 wazuh-modulesd:vulnerability-detector[28862] wm_vuln_detector.c:5228 at wm_vuldet_run_sleep(): DEBUG: Sleeping for 300 seconds...
```
And the output of looking into the database:
```
# for i in {1999..2020}; do echo 'CVEs for '$i':'; sqlite3 /var/ossec/queue/vulnerabilities/cve.db 'select * from NVD_CVE where CVE_ID like "CVE-'$i'-%" limit 1'; done
CVEs for 1999:
CVEs for 2000:
CVEs for 2001:
CVEs for 2002:
CVEs for 2003:
CVEs for 2004:
CVEs for 2005:
CVEs for 2006:
CVEs for 2007:
CVEs for 2008:
CVEs for 2009:
CVEs for 2010:
CVEs for 2011:
CVEs for 2012:
CVEs for 2013:
CVEs for 2014:
CVEs for 2015:
CVEs for 2016:
CVEs for 2017:
CVEs for 2018:
CVEs for 2019:
46522|2019|CVE-2019-0001|CWE-400|Receipt of a malformed packet on MX Series devices with dynamic vlan configuration can trigger an uncontrolled recursion loop in the Broadband Edge subscriber management daemon (bbe-smgd), and lead to high CPU usage and a crash of the bbe-smgd service. Repeated receipt of the same packet can result in an extended denial of service condition for the device. Affected releases are Juniper Networks Junos OS: 16.1 versions prior to 16.1R7-S1; 16.2 versions prior to 16.2R2-S7; 17.1 versions prior to 17.1R2-S10, 17.1R3; 17.2 versions prior to 17.2R3; 17.3 versions prior to 17.3R3-S1; 17.4 versions prior to 17.4R2; 18.1 versions prior to 18.1R3; 18.2 versions prior to 18.2R2.|2019-01-15T21:29Z|2019-10-09T23:43Z
CVEs for 2020:
```

**Comment by DProvinciani** [comment]
Here some details related to what was happening in this scenario and I'm fixing with the PRs [5165](https://github.com/wazuh/wazuh/pull/5165) and [2555](https://github.com/wazuh/wazuh-documentation/pull/2555).

1. There is a bug caused because of the NVD database tables being removed on each iteration for all the indexed years. Basically, this was resulting in a database with only the CVEs of the last year being indexed. This is fixed by [5165](https://github.com/wazuh/wazuh/pull/5165).

2. There is a bug in the NVD related database tables where we were trying to remove a table that didn't exist because of the usage of a bad name. As a result, we got multiples logs telling `ERROR: (5405): SQL error: no such table: NVD_CVE_MATCHES`. This was fixed in 3.13 during the development of the NVD support for Linux agents.

3. We don't support compressed files either for the <url> nor the <path> tag. So, this was specified in the documentation in the PR [2555](https://github.com/wazuh/wazuh-documentation/pull/2555).
